syft/rekor/queryRekor/verify.go

```diff
@@ -0,0 +1,136 @@
+package queryRekor
+
+import (
+	"context"
+	"crypto/sha256"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/anchore/syft/internal/log"
+	"github.com/sigstore/cosign/cmd/cosign/cli/fulcio/fulcioroots"
+	"github.com/sigstore/cosign/pkg/cosign"
+	"github.com/sigstore/rekor/pkg/generated/client"
+	"github.com/sigstore/rekor/pkg/generated/models"
+)
+
+// verify the inclusion proof
+func verifyLogEntry(ctx context.Context, rekorClient *client.Rekor, entry *models.LogEntryAnon) error {
+	err := cosign.VerifyTLogEntry(ctx, rekorClient, entry)
+	if err != nil {
+		log.Debug("Error verifying Rekor entry")
+		return err
+	}
+	return nil
+}
+
+// verify the certificate
+func verifyCert(rekorClient *client.Rekor, cert *x509.Certificate) error {
+
+	certCheckOpts.RekorClient = rekorClient
+	// this code is now in sigstore, not cosign. Will break when cosign dependency updates
+	certCheckOpts.RootCerts = fulcioroots.Get()
+	certCheckOpts.IntermediateCerts = fulcioroots.GetIntermediates()
+
+	_, err := cosign.ValidateAndUnpackCert(cert, certCheckOpts)
+	if err != nil {
+		log.Debug("certificate could not be verified")
+		return err
+	}
+
+	return nil
+}
+
+// return an error if the log entry timestamp is not within the certificate valid time range
+func verifyEntryTimestamp(cert *x509.Certificate, entry *models.LogEntryAnon) error {
+	time := time.Unix(*entry.IntegratedTime, 0)
+	return cosign.CheckExpiry(cert, time)
+}
+
+// verify the attestation hash equals the hash in the rekor entry body.
+// *** This does NOT verify any signature.
+// No error means that the attestation we have and the attestation referenced in rekor are equal, NOT that it is what the user intended to upload.
+// We trust rekor to verify signature ***
+func verifyAttestationHash(encounteredAttestation string, intotoV001 *models.IntotoV001Schema) error {
+
+	attBytes := []byte(encounteredAttestation)
+	hasher := sha256.New()
+	hasher.Write(attBytes)
+	hash := hasher.Sum(nil)
```

Comment:
Given bytes are available, can use the [Sum256](https://pkg.go.dev/crypto/sha256#Sum256) function instead